pkgs/dart_mcp_server/bin/main.dart

```diff
@@ -94,9 +98,38 @@ final argParser =
             'cursor which claim to have roots support but do not actually '
             'support it.',
       )
+      ..addOption(
+        logFileOption,
+        help:
+            'Path to a file to log all MPC protocol traffic to. File will be '
+            'overwritten if it exists.',
+      )
       ..addFlag(help, abbr: 'h', help: 'Show usage text');
 
 const dartSdkOption = 'dart-sdk';
 const flutterSdkOption = 'flutter-sdk';
 const forceRootsFallback = 'force-roots-fallback';
 const help = 'help';
+const logFileOption = 'log-file';
+
+/// Creates a `Sink<String>` for [logFile].
+Sink<String> createLogSink(io.File logFile) {
+  logFile.createSync(recursive: true);
+  final fileByteSink = logFile.openWrite(
+    mode: io.FileMode.write,
+    encoding: utf8,
+  );
+  final logSink = fileByteSink.transform<String>(
+    StreamSinkTransformer.fromHandlers(
+      handleData: (String data, EventSink<List<int>> innerSink) {
+        innerSink.add(utf8.encode(data));
+        // It's a log, so we want to make sure it's always up-to-date.
+        fileByteSink.flush();
+      },
+      handleDone: (EventSink<List<int>> innerSink) {
+        innerSink.close();
+      },
+    ),
+  );
+  return logSink;
```

Comment:
[nit] We can let inference fill in more here

```suggestion
  return fileByteSink.transform(
    StreamSinkTransformer.fromHandlers(
      handleData: (data, innerSink) {
        innerSink.add(utf8.encode(data));
        // It's a log, so we want to make sure it's always up-to-date.
        fileByteSink.flush();
      },
      handleDone: (innerSink) {
        innerSink.close();
      },
    ),
  );
```